data/zmod/quadratic_reciprocity.lean

```diff
@@ -0,0 +1,564 @@
+/-
+Copyright (c) 2018 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes
+-/
+import field_theory.finite data.zmod.basic linear_algebra.prod_module
+
+open function finset nat finite_field zmodp
+
+@[simp] lemma prod_range_id_eq_fact (n : ℕ) : ((range n.succ).erase 0).prod (λ x, x) = fact n :=
```

Comment:
this should be somewhere else. It should be at the place where `fact` is defined, but that's too early.

What about `big_operators`.